beluga-aws-sdk/src/jobs/datetime.rs

```diff
@@ -0,0 +1,40 @@
+use std::ops::Add;
+use std::time::Duration;
+
+use chrono::prelude::*;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Deserializer, Serializer};
+
+pub(super) fn serialize<S>(date: &Option<DateTime<Utc>>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let date = date.map(|date| {
+        let timestamp = date.timestamp(); // Get the timestamp in whole seconds
+        let nanoseconds = date.timestamp_subsec_nanos(); // Get the fractional part (nanoseconds)
+                                                         // Calculate the exact time in seconds with milliseconds precision
+        let fractional_seconds = nanoseconds as f64 / 1_000_000_000_f64; // Convert nanoseconds to fractional seconds
+        timestamp as f64 + fractional_seconds // Combine whole and fractional
+                                              // seconds
+    });
+
+    if let Some(date) = date {
+        serializer.serialize_some::<f64>(&date)
+    } else {
+        serializer.serialize_none()
+    }
+}
+
+pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<Option<DateTime<Utc>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Option::<f64>::deserialize(deserializer).map(|seconds_opt| {
+        seconds_opt.map(|seconds| {
+            DateTime::from_naive_utc_and_offset(
+                NaiveDateTime::UNIX_EPOCH.add(Duration::from_secs_f64(seconds)),
```

Comment:
I believe this will panic on overflow, right? Can we add a unit test for this case and make sure that we don't panic?

Reply:
Yes, having tests is necessary since many places require testing. Let's create a separate task for it. For instance, it would be ideal to have a test broker for the MQTT client to test basic scenarios. What do you think?